source/Nevermore/Advanced/DeadlockAwareLock.cs

```diff
@@ -12,34 +14,53 @@ namespace Nevermore.Advanced
     ///     going to be a deadlock. In other words: very few false positives; probably some false negatives; better than
     ///     nothing.
     /// </summary>
-    public class DeadlockAwareLock : SemaphoreSlim
+    public class DeadlockAwareLock : IDisposable
     {
+        static readonly ILog Log = LogProvider.For<DeadlockAwareLock>();
+        
+        readonly SemaphoreSlim semaphore = new(1, 1);
+        
         int? taskWhichHasAcquiredLock;
         int? threadWhichHasAcquiredLock;
 
-        public DeadlockAwareLock() : base(1, 1)
+        readonly bool logOnConcurrentExecution;
+
+        public DeadlockAwareLock(bool logOnConcurrentExecution)
```

Comment:
minor naming: Perhaps just me, but when I see "logOnConcurrentExecution" I think about logging on to things - e.g. sending username/password to SQL server and such.

Perhaps just `logConcurrentExecution`?

Reply:
That's fair, renamed in e856b5bd93c5da2152ecd261acf018724baa2979